index.js

```diff
@@ -90,11 +98,88 @@ function inject (dispatchFunc, options, callback) {
   }
 }
 
+function Chain (dispatch, option) {
+  this.option = Object.assign({}, option)
+  this.dispatch = dispatch
+  this._hasInvoked = false
+  this._promise = null
+}
+
+const httpMethods = [
+  'delete',
+  'get',
+  'head',
+  'options',
+  'patch',
+  'post',
+  'put',
+  'trace'
+]
+
+httpMethods.forEach(method => {
+  Chain.prototype[method] = function (url) {
+    if (this._hasInvoked === true || this._promise) {
+      throwIfAlreadyInvoked()
```

Comment:
This is not ok, you are missing a return statement.

Reply:
Em...I wonder is there any particular reason that we should add a return statement after `throw`.

Reply:
ok